src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,85 @@
+package duke.task;
+
+import duke.util.Storage;
+
+import java.util.ArrayList;
+
+/**
+ * Class the holds the tasks provided by the user.
+ */
+public class TaskList {
+    
+    private ArrayList<Task> tasks;
+
+    /**
+     * Creates a brand new task list.
+     */
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    /**
+     * Creates a task list from existing data.
+     * @param storage Storage object that will load the data.
+     */
+    public void loadFromStorage(Storage storage) {
+        storage.loadData(this);
+    }
+    
+    public void addTask(Task task) {
+        tasks.add(task);
+    }
+    
+    public void deleteTask(Task task) {
+        tasks.remove(task);
+    }
+    
+    public void markAsDone(Task task) {
+        task.markDone();
+    } 
+    
+    public Task getTask(int taskIdx) {
+        return tasks.get(taskIdx - 1);
+    }
+    
+    public int size() {
+        return tasks.size();
+    }
+
+    /**
+     * Finds and list tasks with description containing the keyword.
+     * @param keyword Keyword for the search.
+     * @return
+     */
+    public String listTasksWithKeyword(String keyword) {
+        StringBuilder output = new StringBuilder();
+        int i = 1;
+        
+        for (Task task : tasks) {
+            if (task.getDescription().contains(keyword)) {
```

Comment:
Maybe consider if you want to differentiate between lowercases and uppercases? And also, if you wish to search for matching elements in the date and time for events and deadlines as well?